cai_causal_graph/graph_components.py

```diff
@@ -199,6 +199,15 @@ def to_dict(self, include_meta: bool = True) -> dict:
 
         return node_dict
 
+    @classmethod
+    def from_dict(cls, node_dict: dict) -> Node:
+        """Return a `cai_causal_graph.graph_components.Node` instance from a dictionary."""
+        return cls(
+            identifier=node_dict['identifier'],
+            variable_type=node_dict['variable_type'],
+            meta=node_dict.get('meta', {}),
```

Comment:
We don't look at variable_type but look at it in `TimeSeriesNode`. Is that missing here?